### DISCUSSION TITLE
Non-primitives as enum values

### DISCUSSION THREAD

**Discussion by kdaniel21** [MichalLytek/type-graphql#826] [Q&A] [answered]
Hey 👋!

I've got a rather simple question: is it safe to use non-primitives as the values of an enum?

In my particular case these would be mongoose models (=objects). I have tried it before asking and it seems to work just fine, so I'm kind of looking for confirmation. Unfortunately I haven't found anything regarding this in the docs, the closest I have is this from the [graphql-js docs](https://graphql.org/graphql-js/type/#graphqlenumtype):

> Some leaf values of requests and input values are Enums. GraphQL serializes Enum values as strings, however internally Enums can be represented by any kind of type, often integers.

Some sample code, in case it's helpful:
```ts
export const MyEnum = {
  Foo: FooModel,
  Bar: BarModel
};

export type MyEnum = typeof MyEnum[keyof typeof MyEnum];

registerEnumType(MyEnum, { name: "MyEnum" });
```

Thanks!

**Comment by MichalLytek** [comment, marked as answer]
TypeScript support only string and number enum members, so you can't use arbitrary things as enum values.

If you cheat and use JS objects as fake enums in `registerEnumType`, be aware that now you're on your own and things can break, especially between updates in the future.

**Reply by kdaniel21** [reply]
Yeah, I cheated by using JS objects. I guess I'm still forced to use JS objects (as the other idea would be to use the model names, but dynamically which doesn't work with enums either), still, at the end of the day it doesn't matter then if the values are strings or non-primitives (with JS objects).

Thanks for your blazing fast answer 🙌!

**Reply by MichalLytek** [reply]
Personally, I would go with simple enum for GraphQL purposes and then a mapper object:
```ts
export const enumToModelMap: Record<MyEnum, Mongoose.Model> = {
  [MyEnum.Foo]: FooModel,
  [MyEnum.Bar]: BarModel,
};

const modelToWorkOn = enumToModelMap[args.enumInput];
```
